### ISSUE TITLE
Issue found on global run but not on single file run

### ISSUE THREAD

**Issue by dvarrazzo** [python/mypy#10031]
Hello,

I have some code with a false positive, approximatively:

```python
    def _stream_fetchone_gen(self) -> Optional["PGresult"]:
            # code, code, code
            self._raise_from_results([res])
            return None  # TODO: shouldn't be needed

    def _raise_from_results(self, results: Sequence["PGresult"]) -> NoReturn:
        raise # something unconditionally
```

As the TODO says, I would have expected the return not being needed. However, the problem is that the issue is reported only in a mypy run with no arguments, and not if that single file is processed.

Steps to reproduce:

```
git clone  git@github.com:psycopg/psycopg3.git
cd psycopg3
git reset --hard 8429be5
pip install mypy==0.800
sed -i "/be needed/d" psycopg3/psycopg3/cursor.py
cd psycopg3
mypy
# psycopg3/cursor.py:257: error: Missing return statement  [return]
# Found 1 error in 1 file (checked 42 source files)
mypy psycopg3/cursor.py 
# Success: no issues found in 1 source file
```

mypy version is 0.800 on Python 3.8.5 on Linux. mypy.ini is included in the repos.
